### ISSUE TITLE
If a story mission requires combat with pirates, use a separate pirate government instead of forcing combat or negative pirate reputation

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#5959]
**Is your feature request related to a problem? Please describe.**
During certain story missions, combat with pirates is required. In such cases, your reputation with pirates is forced to negative. It can be considerably difficult to not only make pirates friendly, but also do that while keeping other factions friendly at the same time, and running into these missions and having them not only force your reputation with pirates to negative but also require you fight pirates and hurt your reputation further in order to complete the mission can be discouraging.

**Related Issue Links**
#4628 

**Describe the solution you'd like**
Create a dummy government for pirates that are required to be killed during missions, akin to the dummy government used for certain independent ships that you are capable of killing. This dummy government is always hostile and doesn't impact your reputation with normal pirates.

**Describe alternatives you've considered**
Create regional pirate governments, and only set the regional government that you need to fight to have a negative reputation when necessary. This contains the impact of requiring that your reputation with pirates be negative by only needing you to mend your reputation with a certain region instead of causing all pirates everywhere to turn hostile. In-universe this also makes a bit more sense; why should the pirates up north get mad that I had to defend myself from some Leviathans attacking me on Thule?

**Additional context**
Story missions that force your reputation negative include but may not be limited to:
* "Wanderers: Sestor Search"

Story missions that require combat with pirates include but may not be limited to:
* "Deep: Mystery Cubes 3/4"
* "Deep: Scientist Rescue 3A"
* "FW Plasma Testing 1B"
* "FW Alphas 1.1/1.2"
* "FW Pirates 4"
* "FW Defend Farpoint"

In some cases you can allow your mission escorts to do the killing for you, but the pirates still all need to be disabled, evaded, or killed in the above missions.

**Comment by Zitchas** [comment]
Having mission required combat with pirates *not* hurt your reputation only makes sense if the pirates involved are specific individuals that for whatever reason are renegades from the usual pirates. 

The whole point of the reputation system is to reflect the opinion of the larger group in response to the player's actions - if we continually make dummy governments and exceptions, pretty soon it will be relegated to simply a tally of how the player acts *outside* of missions, and will be less valuable for that.

I much prefer your alternative: Having a separate government for each major region of pirates makes a lot more sense to me.

As another alternative, we could borrow from EVN, wherein there was more than one pirate government. At least three of them, if I recall. In particular, there was one pirate government that was basically a subset of the others and was basically the extremely desperate/brave "attack everything" pirate that acted kind of like our current one. (in contrast to the regular pirates that were more prone to running away when injured/outgunned and decidedly more interested in loot than they were in killing things). I believe they were called Marauders or Berserkers or something like that. While killing Pirates might impact certain missions and whatnot, Marauders were the generic blood thirsty pirates that are so addicted to slaughtering things that everyone likes to see you kill. Except for the regular pirates, who are neutral about it. 

So instead of having a separate dummy government that just makes reputation and mission completion that much less relevant, I propose that we have two categories of pirate governments:
1) Regular pirates: One or more regional pirate governments that are used when missions should impact the player's reputation among the larger pirate population.
2) Marauders/Reavers: Blood thirsty pirates that just want to kill stuff. This government would be used when we want the pirate opposition in missions to be so reviled that regular pirates aren't upset with the player killing them. If implemented, we should probably convert a percentage of the normal pirate spawns into Marauders. Maybe 20% of them or so.

This also would mean that - even if a player spends the effort to make friends with pirates - they are likely to still be attacked by the Marauders. Which makes sense, since pirates aren't a unified government, but rather, well, pirates. So being friendly with pirates isn't a guaranteed route to universal safety. Also means that there's still someone that players can kill with impunity.

**Comment by LordInsane** [comment]
Yes, EVN had a pirate group called "Marauders" that were distinguished by extreme bravery and aggression, generally low-quality ships, small fleets and popping up pretty much everywhere. It might perhaps be relevant that another of the four pirate factions in EVN was a regional grouping, which due to how the map and game was structured meant that the 'normal' Pirate faction was effectively a regional grouping itself.

**Comment by PeacefulPotato** [comment]
#6543 partially fixes this

**Comment by Hurleveur** [comment]
I made sure the Unfettered campaign doesnt force you to be hostile to pirate governments, using dummy govs at first, but now they've been developped to be much more than that.
Maybe it should always be the case instead if just fighting "pirates", HR does this too.
